src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,36 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+    private static final int MILLIS = 300;
+
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            return CompletableFuture.completedFuture(cache.get(userId));
+        }
+        return CompletableFuture.supplyAsync(() -> getUserDetails(userId), executor);
+    }
+
+    private UserData getUserDetails(String userId) {
+        try {
+            Thread.sleep(MILLIS);
+        } catch (InterruptedException e) {
+            throw new RuntimeException("Failed to retrieve details for userId: " + userId, e);
+        }
+
+        UserData userData = new UserData(userId, "Details for " + userId);
+        cache.put(userId, userData);
```

Comment:
Directly modifying the cache within the `getUserDetails` method can lead to potential issues if the cache policy changes or requires additional logic. It might be beneficial to separate the concerns by having a dedicated method to update the cache.

Reply:
Make it simple stupid